### DISCUSSION TITLE
new default kafkaProjectionUpdates configuration in real time updater service override existing configuration

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#145] [Q&A] [answered]
Hi,
with the new update of the console (version 10.2.0) I found the real time updater with a new configuration about the projection updates.

The problem is that if the real time updater service was already configured to send the projection update messages, the configuration that the service will refer to will be the new one.

The temporary fix I did was to replace the configuration path of the previously set up configuration.

The other problem is that the new configurations doesn't have the edit enabled. Where the config files are declared? How can I write/edit them?

**Comment by seraph-eeno** [comment]
Hi Omar,
I was looking at it too.

As much as I understood, as for the new configurations not being editable, I saw that if you hover with your cursor on top of the _info_ icon next to the name of the Environment Variable, it will show a hint to redirect you to the _Projections_ section. Once there, the _List of Systems of Records_ appears. If you enter in one of those and select a specific projection, you can find at the bottom of the page the section called **_Kafka Topics_**, where at least you can configure the topic maps.
If you want to change the type of the Kafka message instead, inside the same projection view page, there is the tab _**Real Time Updater**_ in which you can specify that.
Strategies and Cast Functions can be configured in the same old sections.

I guess this covers much of the configuration for a Real Time Updater.

**Comment by ThisIsDemetrio** [comment, marked as answer]
Hi @elmalakomar, and thanks for your question.

With v10.2.0 of the console, we've released a new update of the Fast Data where you the _pr-update_ topics are generated by default and applied to a configMap mounted by default in _/home/node/app/kafkaProjectionUpdates_. These definition of these topics, divided by different environment, are included in a Public Variable named _FAST_DATA_PR_UPDATES_MAP_.

It looks like there's a bug on existing System of Records where the configMap dedicated to projection update topics are rewritten if the configMap is saved in _/home/node/app/kafkaProjectionUpdates_. We deeply apologize for this, but we're going to work on it right now.

This is all because we want to simplify the configuration of the Fast Data, automatically creating everything you need, projection updates topic included. But unfortunately it seems that the default folder for this configMap is the same you are currently using.

In case you need (or you simply prefer) to use your own topics, you would create a new configMap in a new folder (in your service, go to the _ConfigMaps_ section and click top _Add ConfigMap_ button, to the right - give a name to the folder and create a new configMap file called _kafkaProjectionUpdates.json_ with your customized topics).

Then, in the _Environment Variables_ section, update the _KAFKA_PROJECTION_UPDATES_FOLDER_ variable with the name of the folder you've just created.

Sorry again for the chaos, but in case you need more explanations or details, feel free to ask more.
Hope it helps
|D|

**Reply by elmalakomar** [reply]
Thanks for the clear answer @ThisIsDemetrio !
This update about the environment variables FAST_DATA_PR_UPDATES_MAP is documented?

**Reply by ThisIsDemetrio** [reply]
Unfortunately, updated documentation about the projection updates and the FAST_DATA_PR_UPDATES_MAP environment variables has not been released with this feature. We're investigating on it, but rest assured we're going to include it in the next update of the docs.
